// @polygon.io/client-js/lib/rest/forex/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.forexClient = void 0;
const request_1 = require("../transport/request");
const aggregates_1 = require("./aggregates");
const aggregatesGroupedDaily_1 = require("./aggregatesGroupedDaily");
const conversion_1 = require("./conversion");
const quotes_1 = require("./quotes");
const lastQuote_1 = require("./lastQuote");
const previousClose_1 = require("./previousClose");
const snapshots_1 = require("./snapshots");
const forexClient = (apiKey, apiBase = "https://api.polygon.io") => ({
    aggregates: (0, request_1.auth)(apiKey, aggregates_1.aggregates, apiBase),
    aggregatesGroupedDaily: (0, request_1.auth)(apiKey, aggregatesGroupedDaily_1.aggregatesGroupedDaily, apiBase),
    conversion: (0, request_1.auth)(apiKey, conversion_1.conversion, apiBase),
    quotes: (0, request_1.auth)(apiKey, quotes_1.quotes, apiBase),
    lastQuote: (0, request_1.auth)(apiKey, lastQuote_1.lastQuote, apiBase),
    previousClose: (0, request_1.auth)(apiKey, previousClose_1.previousClose, apiBase),
    snapshotAllTickers: (0, request_1.auth)(apiKey, snapshots_1.snapshotAllTickers, apiBase),
    snapshotGainersLosers: (0, request_1.auth)(apiKey, snapshots_1.snapshotGainersLosers, apiBase),
    snapshotTicker: (0, request_1.auth)(apiKey, snapshots_1.snapshotTicker, apiBase),
});
exports.forexClient = forexClient;
exports.default = exports.forexClient;
