// @polygon.io/client-js/lib/rest/stocks/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.stocksClient = void 0;
const request_1 = require("../transport/request");
const aggregates_1 = require("./aggregates");
const aggregatesGroupedDaily_1 = require("./aggregatesGroupedDaily");
const dailyOpenClose_1 = require("./dailyOpenClose");
const previousClose_1 = require("./previousClose");
const lastQuote_1 = require("./lastQuote");
const lastTrade_1 = require("./lastTrade");
const snapshots_1 = require("./snapshots");
const quotes_1 = require("./quotes");
const trades_1 = require("./trades");
const stocksClient = (apiKey, apiBase = "https://api.polygon.io") => ({
    aggregates: (0, request_1.auth)(apiKey, aggregates_1.aggregates, apiBase),
    aggregatesGroupedDaily: (0, request_1.auth)(apiKey, aggregatesGroupedDaily_1.aggregatesGroupedDaily, apiBase),
    dailyOpenClose: (0, request_1.auth)(apiKey, dailyOpenClose_1.dailyOpenClose, apiBase),
    lastQuote: (0, request_1.auth)(apiKey, lastQuote_1.lastQuote, apiBase),
    lastTrade: (0, request_1.auth)(apiKey, lastTrade_1.lastTrade, apiBase),
    previousClose: (0, request_1.auth)(apiKey, previousClose_1.previousClose, apiBase),
    quotes: (0, request_1.auth)(apiKey, quotes_1.quotes, apiBase),
    snapshotAllTickers: (0, request_1.auth)(apiKey, snapshots_1.snapshotAllTickers, apiBase),
    snapshotGainersLosers: (0, request_1.auth)(apiKey, snapshots_1.snapshotGainersLosers, apiBase),
    snapshotTicker: (0, request_1.auth)(apiKey, snapshots_1.snapshotTicker, apiBase),
    trades: (0, request_1.auth)(apiKey, trades_1.trades, apiBase),
});
exports.stocksClient = stocksClient;
exports.default = exports.stocksClient;
