// @polygon.io/client-js/lib/rest/options/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.optionsClient = exports.snapshotOptionContract = void 0;
const request_1 = require("../transport/request");
const aggregates_1 = require("./aggregates");
const dailyOpenClose_1 = require("./dailyOpenClose");
const previousClose_1 = require("./previousClose");
const trades_1 = require("./trades");
const lastTrade_1 = require("./lastTrade");
const quotes_1 = require("./quotes");
const snapshots_1 = require("./snapshots");
var snapshots_2 = require("./snapshots");
Object.defineProperty(exports, "snapshotOptionContract", { enumerable: true, get: function () { return snapshots_2.snapshotOptionContract; } });
const optionsClient = (apiKey, apiBase = "https://api.polygon.io") => ({
    aggregates: (0, request_1.auth)(apiKey, aggregates_1.aggregates, apiBase),
    dailyOpenClose: (0, request_1.auth)(apiKey, dailyOpenClose_1.dailyOpenClose, apiBase),
    previousClose: (0, request_1.auth)(apiKey, previousClose_1.previousClose, apiBase),
    trades: (0, request_1.auth)(apiKey, trades_1.trades, apiBase),
    lastTrade: (0, request_1.auth)(apiKey, lastTrade_1.lastTrade, apiBase),
    quotes: (0, request_1.auth)(apiKey, quotes_1.quotes, apiBase),
    snapshotOptionContract: (0, request_1.auth)(apiKey, snapshots_1.snapshotOptionContract, apiBase),
});
exports.optionsClient = optionsClient;
exports.default = exports.optionsClient;
