// @polygon.io/client-js/lib/rest/crypto/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.cryptoClient = void 0;
const request_1 = require("../transport/request");
const aggregates_1 = require("./aggregates");
const aggregatesGroupedDaily_1 = require("./aggregatesGroupedDaily");
const dailyOpenClose_1 = require("./dailyOpenClose");
const trades_1 = require("./trades");
const lastTrade_1 = require("./lastTrade");
const previousClose_1 = require("./previousClose");
const snapshots_1 = require("./snapshots");
const cryptoClient = (apiKey, apiBase = "https://api.polygon.io") => ({
    aggregates: (0, request_1.auth)(apiKey, aggregates_1.aggregates, apiBase),
    aggregatesGroupedDaily: (0, request_1.auth)(apiKey, aggregatesGroupedDaily_1.aggregatesGroupedDaily, apiBase),
    dailyOpenClose: (0, request_1.auth)(apiKey, dailyOpenClose_1.dailyOpenClose, apiBase),
    lastTrade: (0, request_1.auth)(apiKey, lastTrade_1.lastTrade, apiBase),
    trades: (0, request_1.auth)(apiKey, trades_1.trades, apiBase),
    previousClose: (0, request_1.auth)(apiKey, previousClose_1.previousClose, apiBase),
    snapshotAllTickers: (0, request_1.auth)(apiKey, snapshots_1.snapshotAllTickers, apiBase),
    snapshotGainersLosers: (0, request_1.auth)(apiKey, snapshots_1.snapshotGainersLosers, apiBase),
    snapshotTicker: (0, request_1.auth)(apiKey, snapshots_1.snapshotTicker, apiBase),
    snapshotTickerFullBookL2: (0, request_1.auth)(apiKey, snapshots_1.snapshotTickerFullBookL2, apiBase),
});
exports.cryptoClient = cryptoClient;
exports.default = exports.cryptoClient;
