// @polygon.io/client-js/lib/rest/reference/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.referenceClient = void 0;
const request_1 = require("../transport/request");
const conditions_1 = require("./conditions");
const exchanges_1 = require("./exchanges");
const marketHolidays_1 = require("./marketHolidays");
const marketStatus_1 = require("./marketStatus");
const optionsContract_1 = require("./optionsContract");
const optionsContracts_1 = require("./optionsContracts");
const dividends_1 = require("./dividends");
const stockSplits_1 = require("./stockSplits");
const stockFinancials_1 = require("./stockFinancials");
const tickerDetails_1 = require("./tickerDetails");
const tickerNews_1 = require("./tickerNews");
const tickers_1 = require("./tickers");
const tickerTypes_1 = require("./tickerTypes");
const referenceClient = (apiKey, apiBase = "https://api.polygon.io") => ({
    conditions: (0, request_1.auth)(apiKey, conditions_1.conditions, apiBase),
    exchanges: (0, request_1.auth)(apiKey, exchanges_1.exchanges, apiBase),
    marketHolidays: (0, request_1.auth)(apiKey, marketHolidays_1.marketHolidays, apiBase),
    marketStatus: (0, request_1.auth)(apiKey, marketStatus_1.marketStatus, apiBase),
    optionsContract: (0, request_1.auth)(apiKey, optionsContract_1.optionsContract, apiBase),
    optionsContracts: (0, request_1.auth)(apiKey, optionsContracts_1.optionsContracts, apiBase),
    dividends: (0, request_1.auth)(apiKey, dividends_1.stockDividends, apiBase),
    stockSplits: (0, request_1.auth)(apiKey, stockSplits_1.stockSplits, apiBase),
    stockFinancials: (0, request_1.auth)(apiKey, stockFinancials_1.stockFinancials, apiBase),
    tickerDetails: (0, request_1.auth)(apiKey, tickerDetails_1.tickerDetails, apiBase),
    tickerNews: (0, request_1.auth)(apiKey, tickerNews_1.tickerNews, apiBase),
    tickers: (0, request_1.auth)(apiKey, tickers_1.tickers, apiBase),
    tickerTypes: (0, request_1.auth)(apiKey, tickerTypes_1.tickerTypes, apiBase),
});
exports.referenceClient = referenceClient;
exports.default = exports.referenceClient;
